### ISSUE TITLE
Problem with generated request XML

### ISSUE THREAD

**Issue by luismonge79** [WsdlToPhp/PackageGenerator#56]
Hello.
I'm having some trouble using this lib and I hope someone can help me.

Using the lib Service SOAP client, the generated XML is the following and the request fails:
`<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="urn://XXX/AppEntityLayer/Services/CustomerContract/v3" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <SOAP-ENV:Body>
    <ns1:GetDetailsRequest>
      <Header xsi:type="ns1:Header"/>
      <Authentication>
        <UserID>user123</UserID>
        <Language>PT</Language>
      </Authentication>
      <CustomerContract xsi:type="ns1:CustomerContract"/>
    </ns1:GetDetailsRequest>
  </SOAP-ENV:Body>
</SOAP-ENV:Envelope>`

Using the native PHP SOAP client, I get the following generated XML and it works:
`<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="urn://XXX/AppEntityLayer/Services/CustomerContract/v3">
  <SOAP-ENV:Body>
    <ns1:GetDetailsRequest>
      <Header>
        <application>App</application>
        <target>CRM</target>
      </Header>
      <Authentication>
        <UserID>user123</UserID>
        <Language>PT</Language>
      </Authentication>
      <CustomerContract>
        <NIF>PT123456789</NIF>
      </CustomerContract>
    </ns1:GetDetailsRequest>
  </SOAP-ENV:Body>
</SOAP-ENV:Envelope>`

I could use the native SOAP client, but I don't really know why, the wsdl_classmap is not working. The response isn't mapping to the Structs objects. I'm using the ClassMap::get() method.
PHP version is 5.5
If more information is needed, I prefer not doing publicly.

Thanks in advance!

**Comment by GroxExMachine** [comment]
Hello.

Probably, you use static method ClassMap::get() where you need to use method from corresponding object.

Did you checked tutorial.php?

Example:
```php
/**
 * Samples for Get ServiceType
 */
$get = new \AppBundle\Provider\YandexDirect\AdExtensions\ServiceType\Get($options);
/**
 * Sample call for get operation/method
 */
if ($get->get(new \AppBundle\Provider\YandexDirect\AdExtensions\StructType\GetRequest()) !== false) {
    print_r($get->getResult());
} else {
    print_r($get->getLastError());
}
```
And, of course, you should setup Headers, looks like this 
```php
$get->setHttpHeader('application', 'App');
$get->setHttpHeader('target', 'CRM');
```

**Comment by mikaelcom** [comment]
Could you share at contact@wsdltophp.com your scripts, the one using the native PHP SoapClient class and the one using the generated package?

Thx

**Comment by mikaelcom** [comment]
Hi,

Thanks to our private conversation, I'm now able to know where the issue comes from.

First, I faced this issue before and fixed it by overriding the native SoapClient::__doRequest method in order to fix the XML request. In your case I digged a little more,
Second, I thought the generated classmap array was correct,
Third, the issue seems to come from the fact that the PHP class names in the generated classmap corresponding to the XSD type must contain the full namespace (which is the case) with the first antislash (which is not the case). I thought I checked this at the very beginning but it seems I might be wrong.

If anyone has a answer with reference I'm very interested, in the meantime, I'm going to do my research then ensure this is fixed once for all.

Thanks for your patience and your posted issue,
Regards,

**Comment by mikaelcom** [comment]
I remember having read this bug report, which is contradicting, at https://bugs.php.net/bug.php?id=69280.

But the last comment at https://bugs.php.net/bug.php?id=54074 seems to indicate the contrary (meaning that prefixing \ is required).

It seems to be dependent on the PHP version. So I must do my tests on different PHP versions first

**Comment by mikaelcom** [comment]
After having done tests with a script using fully qualified name class with namespace such as `\\Structs\\MyItemRequest` in **PHP 5.4**, **PHP 5.5**, **PHP 5.6** and **PHP 7.0**, it works perfectly.

This means I have to fix the classmap generation by always specifying the fully qualified class name,

Any objection in the audience?

**Comment by GroxExMachine** [comment]
Nope.

**Comment by mikaelcom** [comment]
Hi,

It should be fixed now using [1.6.0](https://github.com/WsdlToPhp/PackageGenerator/releases/tag/1.6.0) or [2.1.0](https://github.com/WsdlToPhp/PackageGenerator/releases/tag/2.1.0),

Good luck,

**Comment by luismonge79** [comment]
I confirm that it's fixed. Thanks so very much.
